nano/node/online_reps.cpp

```diff
@@ -0,0 +1,99 @@
+#include <nano/node/node.hpp>
+#include <nano/node/online_reps.hpp>
+
+#include <cassert>
+
+nano::online_reps::online_reps (nano::node & node_a, nano::uint128_t minimum_a) :
```

Comment:
I wonder if during these refactorings we could change what we pass to these classes rather than `node` so it's clearer what dependencies it has (as we have quite a few initialization issues with `node`). For instance this class just needs `ledger` and `max_weight_samples`. Doesn't have to be done now, but something to consider in the future.

Reply:
Yeah we can do a separate PR that goes through these injections